### ISSUE TITLE
Take advantage of built in support for full page screenshots

### ISSUE THREAD

**Issue by nextlevelbeard** [webdriverio/visual-testing#622]
**Is your feature request related to a problem? Please describe.**
Often, the full page screenshots are unreliable and contain ghosting of elements due to scrolling effects. 
For example: see https://github.com/webdriverio/visual-testing/issues/125

**Describe the solution you'd like**
Multiple browsers have built in support for this that deliver a clean and ghost free image.

For Firefox, we can create a client side script that calls the `:screenshot` command. See docs here:
https://firefox-source-docs.mozilla.org/devtools-user/taking_screenshots/index.html#taking-screenshots-with-the-web-console

Then execute it with the executor function in `getFullPageScreenshotsDataDesktop`:
https://github.com/webdriverio/visual-testing/blob/b40be9ddd0b8a370ac50c1e49cd6006a5225ca41/packages/webdriver-image-comparison/src/methods/screenshots.ts#L352

**Describe alternatives you've considered**

We could also call the webdriverIO API: `browser.fullPageScreenshot()`, if the `browser` is available to us.
https://webdriver.io/docs/api/gecko/#fullpagescreenshot

**Comment by christian-bromann** [comment]
We should update the `saveScreenshot` command in WebdriverIO core to use WebDriver Bidi primitives for [taking screenshots](https://w3c.github.io/webdriver-bidi/#command-browsingContext-captureScreenshot). They provide full flexibility including full page screenshots.

**Comment by wswebcreation** [comment]
Thanks for your FR @nextlevelbeard , I've investigated the Firefox one in the past and wasn't fully reliable. But I can do a research again

@christian-bromann That would be great! Adding a flag for fullpage might be very useful

**Comment by nextlevelbeard** [comment]
@wswebcreation @christian-bromann 

Am I correct to say that this is now unblocked by PR https://github.com/webdriverio/webdriverio/pull/14267 being merged and released with webdriverIO [v9.12.0](https://github.com/webdriverio/webdriverio/releases/tag/v9.12.0) ?

**Comment by wswebcreation** [comment]
@nextlevelbeard 

Yes, you are correct, but....., there is a "hard dependency" with using the correct version of the `webdriverio` package. Just checking if it's BiDi is not enough. For now, the module doesn't know which `webdriverio` version the user uses. 
The project-committers are now looking for a good solution that is not only usable for the visual package, but also for future packages/modules
